components/encryption/src/manager/mod.rs

```diff
@@ -21,22 +22,34 @@ use crate::{Error, Result};
 
 const KEY_DICT_NAME: &str = "key.dict";
 const FILE_DICT_NAME: &str = "file.dict";
+const ROTATE_PERIOD: u64 = 600;
 
 struct Dicts {
-    file_dict: FileDictionary,
-    key_dict: KeyDictionary,
+    // Maps data file paths to key id and metadata. This file is stored as plaintext.
+    file_dict: Mutex<FileDictionary>,
+    // To lock the writing of file dictionary.
+    file_lock: Mutex<()>,
+    // Maps data key id to data keys, together with metadata. Also stores the data
+    // key id used to encrypt the encryption file dictionary. The content is encrypted
+    // using master key.
+    key_dict: Mutex<KeyDictionary>,
+    // Thread-safe version of current_key_id. Only when writing back to key_dict,
+    // write it back to `key_dict`.
```

Comment:
```suggestion
    // write it back to `key_dict`. Reader should always use this atomic, 
    // instead of key_dict.current_key_id, since the latter can reflect an update-in-progress key.
```